chopsticks-sample/src/test/scala/dev/chopsticks/sample/dstreams/DstreamsMasterWorkerTest.scala

```diff
@@ -0,0 +1,143 @@
+    val cfg = ConfigFactory.load().withValue("iz-logging.level", ConfigValueFactory.fromAnyRef("Crit"))
+    if (!cfg.getBoolean("akka.stream.materializer.debug.fuzzing-mode")) {
+      throw new IllegalArgumentException(
+        "akka.stream.materializer.debug.fuzzing-mode is not 'on' for testing, config loading is not working properly?"
+      )
+    }
+    cfg
+  }
+
+  override def spec: ZSpec[LiveRunnableEnv, Any] = {
+    suite("Master worker suite")(
+      testM("Master worker test") {
+        val managedZio = for {
+          manageServerResult <- DstreamsSampleMasterApp.manageServer
+          (serverBinding, dstreamState) = manageServerResult
+          port = serverBinding.localAddress.getPort
+          lastValue <- ZRef.make(BigInt(0)).toManaged_
+        } yield {
+          import DstreamsSampleWorkerApp.runWorker
+          val assertCurrentValueIsBiggerThanLastSeen =
+            UIO(DstreamsSampleMasterApp.currentValue.get).flatMap { current =>
+              assertM(lastValue.get)(isLessThan(current)) <* lastValue.set(current)
+            }
+          val assertCurrentValueIsTheSameAsLastSeen =
+            UIO(DstreamsSampleMasterApp.currentValue.get).flatMap { current =>
+              assertM(lastValue.get)(equalTo(current))
+            }
+          for {
+            forkedResult <- DstreamsSampleMasterApp.calculateResult(dstreamState).log("Test: calculating result").fork
+
+            _ <- runWorker(port, 1, willCrash = false, willFail = false)
+            _ <- assertCurrentValueIsBiggerThanLastSeen
+
+            _ <- assertM {
+              runWorker(port, 2, willCrash = true, willFail = false)
+                .unit
+                .either
+                .map(_.fold(Failure(_), Success(_)))
+            }(isFailure)
+            _ <- assertCurrentValueIsTheSameAsLastSeen
+
+            _ <- assertM {
+              runWorker(port, 3, willCrash = false, willFail = true)
+                .unit
+                .either
+                .map(_.fold(Failure(_), Success(_)))
+            }(isFailure)
+            _ <- assertCurrentValueIsTheSameAsLastSeen
+
+            _ <- {
+              ZIO
+                .foreachPar_(1 to 4) { i => runWorker(port, 10 + i, willCrash = true, willFail = false).ignore }
+                .fork
+            }
+            _ <- {
+              ZIO
+                .foreachPar_(1 to 4) { i => runWorker(port, 20 + i, willCrash = false, willFail = true).ignore }
+                .fork
+            }
+            _ <- ZIO.foreachPar_(1 to 4) { i => runWorker(port, 30 + i, willCrash = false, willFail = false).ignore }
+            _ <- assertCurrentValueIsBiggerThanLastSeen
+
+            result <- forkedResult.join
+          } yield assert(result)(equalTo(masterConfig.expected))
+        }
+        managedZio.use(identity)
+      } @@ timeout(20.seconds.toJava)
+    ).provideSomeLayer[LiveRunnableEnv].apply(extraLayer.mapError(TestFailure.fail))
+  }
+
+  private def extraLayer = {
```

Comment:
Can we leverage / unify `AkkaDiApp` here for building layers as well as bootstrapping actor system instead?

Reply:
There is now `AkkaDiRunnableSpec` that reuses whatever we can from `AkkaDiApp` and can be used for Akka and DiStage based integration tests

Reply:
@nktpro please let me know if `AkkaDiRunnableSpec` should take type parameter (that would be a config) and whether `diTestM` should accept additional customize method: `customizeConfig: TestConfig => TestConfig`